php/src/Google/Protobuf/Internal/MapField.php

```diff
@@ -131,7 +131,7 @@ public function getLegacyValueClass()
      *
      * This will also be called for: $ele = $arr[$key]
      *
-     * @param int|bool|string $key The key of the element to be fetched.
+     * @param int|string $key The key of the element to be fetched.
```

Comment:
Circling back to this! Sorry for the delay.

The param type here should be the same as the other `@param $key` types (for instance, in `offsetSet`, as it's the same array operation for all of them (php primitive array access). So we can either (1) update the rest of them to be `int|bool|string`, or (2) update this one to be `int|string`.

Advantages of (1): It's technically correct (PHP allows it), and it's "safer" (opening up the types instead of restricting them), as restricting this type _could_ lead to static analysis errors in existing code (Note: As this is static analysis, it isn't BC-breaking, as running code would remain unaffected). 

Advantages of (2): PHP is moving towards stricter types, and when a `bool` is used as a PHP array key, it is casted into an `int` (`0` for `false` and `1` for `true`). So while it's accepted, it isn't truly a valid array key type. If we document this as `int|string`, it's better practice, and also reflects what we would likely do once this library becomes PHP 8.0+ and we use union types. 

Whichever we decide is okay with me! I am happy to change all of them to `int|bool|string`, as it would be the more conservative move!